### PR TITLE
Fix THTensor_(take) negative index check

### LINKED ISSUE #6472: Advanced indexing doesn't validate negative indexes (regression)
Minimum example:

    torch.zeros(10)[torch.LongTensor([-11])] # returns bogus result on 0.4, error on 0.3.1

These all fail as they should:

    torch.zeros(10)[10]
    torch.zeros(10)[-11]
    torch.zeros(10)[torch.LongTensor([10])]

Works correctly on 0.3.1.post2.
Doesn't work on 0.4.0a0+df039e2.

Compiled with:
```
⟫ gcc-5 -v
Using built-in specs.
COLLECT_GCC=gcc-5
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/5/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 5.5.0-8ubuntu1' --with-bugurl=file:///usr/share/doc/gcc-5/README.Bugs --enable-languages=c,ada,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-5 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-5-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-5-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-5-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 5.5.0 20171010 (Ubuntu 5.5.0-8ubuntu1) 

```

### PR DESCRIPTION
**Pull request by ssnl**
Fixes #6472 .

Previously we save invalid indices in `invalidIdx` and checks if `invalidIdx < 0` after OMP loop. This in incorrect because the index can be negative. This PR changes to saving the invalid dimension.

### CODE REVIEW

File: aten/src/TH/generic/THTensorMath.c [hunk 1]
```diff
@@ -450,7 +450,7 @@ void THTensor_(take)(THTensor *r_, THTensor *src, THLongTensor *index)
   // Exceptions must not be thrown across OpenMP parallel sections, so we
   // record the value of the invalid index and throw the exception after the
   // loop.
-  int64_t invalidIdx = -1;
+  int64_t invalidIdxDim = -1;
```
Comment (colesbury) [inline aten/src/TH/generic/THTensorMath.c]: The naming is confusing because this isn't a dim (axis). Maybe invalidIdxOffset or invalidIdxPos

**Review by colesbury — APPROVED** [review]
